File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) 
    {
        if(nums.size() == 0 || nums.size() == 1) return nums.size();

        set<int> st;
        for(auto it: nums) st.insert(it);

        int prev = INT_MIN, res = 1, c = 1;
        for(auto it: st)
        {
            if(prev == INT_MIN) prev = it;
            else
            {
                if(prev + 1 == it) c++;
                else c = 1;
                prev = it;
                res = max(res, c);
            }
        }
        return res;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        hashmap = {}
        res = 0

        for num in nums:
            hashmap[num] = True

        for num in hashmap:
            if num - 1 not in hashmap:
                length = 0
                current = num

                while current in hashmap:
                    length += 1
                    current += 1

                res = max(res, length)

        return res
        